test/bindings/dom.js

```diff
@@ -80,3 +83,96 @@ describe('getAttributes', function() {
     assert.deepEqual(ret.args, l10nArgs);
   });
 }); 
+
+describe.skip('translateFragment', function() {
+  it('should translate fragment', function() {
+    let dom = new L20n.LocalizationObserver();
+    const div = document.createElement('div');
+    div.innerHTML = `
+      <p data-l10n-id="key1"></p>
+      <p data-l10n-id="key2"></p>
+      <div>
+        <p data-l10n-id="key3"></p>
+      </div>
+    `;
+
+    function requestBundles() {
+      let rb = new L20n.ResourceBundle('en-US', [
+        './locales/test.ftl'
+      ], {
+        './locales/test.ftl': `
+key1 = Value 1
+key2 = Value 2
+key3 = Value 3
+      `
+      });
+      return Promise.resolve([
+        rb
+      ]);
+    }
+
+    function createContext(lang) {
+      return new L20n.MessageContext();
+    }
+    const loc = new L20n.HTMLLocalization(requestBundles, createContext);
+    dom.set('main', loc);
+    return dom.translateFragment(div).then(() => {
+      dom.localizations.delete('main');
+      assert.equal(div.getElementsByTagName('p')[0].textContent, 'Value 1');
+      assert.equal(div.getElementsByTagName('p')[1].textContent, 'Value 2');
+      assert.equal(div.getElementsByTagName('p')[2].textContent, 'Value 3');
+    });
+  });
+});
+
+describe('MutationObserver', function() {
+  it('should translate injected element', function(done) {
+    const div = document.createElement('div');
+    let dom = new L20n.LocalizationObserver();
+
+    let mo = new MutationObserver(function(mutations) {
+      let node = mutations[0].addedNodes[0];
+      if (node.nodeType === Node.TEXT_NODE &&
+          node.textContent === 'Value 4') {
+        mo.disconnect(div);
+        dom.disconnectRoot(div);
+        dom.localizations.delete('main');
+        done();
+      }
+    });
+
+    function requestBundles() {
+      let rb = new L20n.ResourceBundle('en-US', [
+        './locales/test.ftl'
+      ], {
+        './locales/test.ftl': `
+key1 = Value 4
+      `
+      });
+      return Promise.resolve([
+        rb
+      ]);
+    }
+
+    function createContext(lang) {
+      return new L20n.MessageContext();
+    }
+    const loc = new L20n.HTMLLocalization(requestBundles, createContext);
+    dom.set('main', loc);
+
+    dom.observeRoot(div);
+
+    mo.observe(div, {
```

Comment:
Instead of creating a new `MutationObserver` here, maybe we should consider emitting events on `document` when mutation records are translated?  I think we'll end up having to do it anyways so this might be a good opportunity to add this and make testing easier.  I can file a bug for this.  Or, if you prefer, we can land this test as-is and then change it.